src/main/java/de/zalando/aruha/nakadi/service/CursorsCommitService.java

```diff
@@ -0,0 +1,50 @@
+package de.zalando.aruha.nakadi.service;
+
+import de.zalando.aruha.nakadi.domain.Cursor;
+import de.zalando.aruha.nakadi.exceptions.NoSuchSubscriptionException;
+import de.zalando.aruha.nakadi.exceptions.ServiceUnavailableException;
+import de.zalando.aruha.nakadi.repository.TopicRepository;
+import de.zalando.aruha.nakadi.repository.zookeeper.ZooKeeperHolder;
+import de.zalando.aruha.nakadi.repository.zookeeper.ZooKeeperLockFactory;
+
+import java.nio.charset.Charset;
+
+import static de.zalando.aruha.nakadi.repository.zookeeper.ZookeeperUtils.runLocked;
+import static java.text.MessageFormat.format;
+
+public class CursorsCommitService {
+
+    private static final Charset CHARSET = Charset.forName("UTF-8");
+
+    private final ZooKeeperHolder zkHolder;
+    private final TopicRepository topicRepository;
+    private final ZooKeeperLockFactory zkLockFactory;
+
+    public CursorsCommitService(final ZooKeeperHolder zkHolder, final TopicRepository topicRepository,
+                                final ZooKeeperLockFactory zkLockFactory) {
+        this.zkHolder = zkHolder;
+        this.topicRepository = topicRepository;
+        this.zkLockFactory = zkLockFactory;
+    }
+
+    public boolean commitCursor(final String subscriptionId, final String eventType, final Cursor cursor)
+            throws ServiceUnavailableException, NoSuchSubscriptionException {
+
```

Comment:
I actually also thought about that possibility but then I changed my mind.
`commitCursor` is designed to commit only one cursor. Validating single cursor is not effective because to validate a cursor you need to get a list of partitions from kafka. So if you need to validate 10 cursors you will need to have 10 calls to kafka.
That's why I decided to validate all cursors together before the call of `commitCursors`